ml/rl/workflow/identify_types_flow.py

```diff
@@ -0,0 +1,135 @@
+        available_features = set(params.keys())
+        assert whitelist_feature_set == available_features, (
+            "Could not identify preprocessing type for these features: {}; "
+            "extra features: {}".format(
+                whitelist_feature_set - available_features,
+                available_features - whitelist_feature_set,
+            )
+        )
+
+    def process(rows: List) -> Dict[int, normalization.NormalizationParameters]:
+        params = dict()
+        for row in rows:
+            assert "feature_name" in row
+            assert "feature_values" in row
+            norm_metdata = normalization.get_feature_norm_metadata(
+                row["feature_name"], row["feature_values"], norm_params
+            )
+            if norm_metdata is not None and (
+                not whitelist_features or row["feature_name"] in whitelist_features
+            ):
+                params[row["feature_name"]] = norm_metdata
+
+        if assert_whitelist_feature_coverage:
+            validate_whitelist_features(params)
+        return params
+
+    return process
+
+
+def identify_normalization_parameters(
+    table_spec: TableSpec,
+    column_name: str,
+    preprocessing_options: PreprocessingOptions,
+    seed: int,
+) -> Dict[int, normalization.NormalizationParameters]:
+    """ Get normalization parameters """
+
+    spark = get_spark_session()
+    df = spark.sql(f"SELECT * FROM {table_spec.table_name}")
+    df = create_normalization_spec_spark(
+        df, column_name, preprocessing_options.num_samples, seed
+    )
+    rows = df.collect()
+    spark.stop()
+
+    normalization_processor = normalization_helper(
+        max_unique_enum_values=preprocessing_options.max_unique_enum_values,
+        quantile_size=preprocessing_options.quantile_size,
+        quantile_k2_threshold=preprocessing_options.quantile_k2_threshold,
+        skip_box_cox=preprocessing_options.skip_box_cox,
+        skip_quantiles=preprocessing_options.skip_quantiles,
+        feature_overrides=preprocessing_options.feature_overrides,
+        whitelist_features=preprocessing_options.whitelist_features,
+        assert_whitelist_feature_coverage=preprocessing_options.assert_whitelist_feature_coverage,
+    )
+    return normalization_processor(rows)
+
+
+def create_normalization_spec_spark(df, column, num_samples: int, seed: int):
+    """Returns approximately num_samples random rows from column of df."""
+
+    df = df.select(
+        explode(col(column).alias("features")).alias("feature_name", "feature_value")
+    )
+
+    # calculate fractions
+    counts_df = df.groupBy("feature_name").count()
+    frac = dict()
+    for row in counts_df.collect():
+        assert num_samples <= row["count"]
+        frac[row["feature_name"]] = num_samples / row["count"]
+
+    # perform sampling and collect them
+    df = df.sampleBy("feature_name", fractions=frac, seed=seed)
```

Comment:
Could you add # TODO with task number and/or github issue to note that we should make this reservoir sampling?

```
# TODO(T1234, #1111): make this reservoir sampling
```